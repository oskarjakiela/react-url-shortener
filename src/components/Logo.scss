.Shorty-Logo {
  margin: 0;
  font-family: 'Montserrat', sans-serif;
  font-weight: 600;
  font-size: 47px;
  text-decoration: underline;
  color: #eb4a42;

  a {
    color: inherit;
  }
}
