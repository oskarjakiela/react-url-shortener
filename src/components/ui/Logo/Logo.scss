.Shorty-Logo {
  margin-top: 0;
  margin-bottom: 0;
  font-family: 'Montserrat', sans-serif;
  font-weight: 600;
  font-size: 47px;
  text-decoration: underline;
  color: #eb4a42;

  a {
    color: inherit;
  }
}
