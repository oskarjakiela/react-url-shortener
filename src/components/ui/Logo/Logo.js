import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

import './Logo.css';


const displayName = 'Shorty-Logo';

const Logo = ({
  children,
  className,
  ...props
}) => (
  <h1 className={classnames(displayName, className)} {...props}>
    <a href="/" title={`${children} Logo`}>
      {children}
    </a>
  </h1>
);

Logo.displayName = displayName;

Logo.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Logo;
