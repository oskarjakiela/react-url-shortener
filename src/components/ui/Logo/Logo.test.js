import React from 'react';
import renderer from 'react-test-renderer';

import Logo from './Logo';


it('<Logo /> renders correctly', () => {
  const tree = renderer
    .create(<Logo>Lorem ipsum</Logo>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('<Logo className="foo-bar" /> renders correctly', () => {
  const tree = renderer
    .create(<Logo className="foo-bar">Lorem ipsum</Logo>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('<Logo title="foo-bar" disabled /> renders correctly', () => {
  const tree = renderer
    .create(<Logo className="foo-bar">Lorem ipsum</Logo>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
