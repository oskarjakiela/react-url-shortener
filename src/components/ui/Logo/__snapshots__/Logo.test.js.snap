// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> renders correctly 1`] = `
<h1
  className="Shorty-Logo"
>
  <a
    href="/"
    title="Lorem ipsum Logo"
  >
    Lorem ipsum
  </a>
</h1>
`;

exports[`<Logo className="foo-bar" /> renders correctly 1`] = `
<h1
  className="Shorty-Logo foo-bar"
>
  <a
    href="/"
    title="Lorem ipsum Logo"
  >
    Lorem ipsum
  </a>
</h1>
`;

exports[`<Logo title="foo-bar" disabled /> renders correctly 1`] = `
<h1
  className="Shorty-Logo"
  disabled={true}
  title="foo-bar"
>
  <a
    href="/"
    title="Lorem ipsum Logo"
  >
    Lorem ipsum
  </a>
</h1>
`;
